app/battle/battle_manager.py

```diff
@@ -0,0 +1,56 @@
+from app.knights.knight import Knight
+from app.equipment.armour import Armour
+from app.equipment.weapon import Weapon
+from app.equipment.potion import Potion
+
+
+class BattleManager:
+    def __init__(self, knights_config: tuple) -> None:
+        self.knights = [
+            Knight(
+                knight_data["name"],
+                knight_data["power"],
+                knight_data["hp"],
+                [
+                    Armour(armour["part"],
+                           armour["protection"])
+                    for armour in knight_data["armour"]
+                ],
+                Weapon(knight_data["weapon"]["name"],
+                       knight_data["weapon"]["power"]),
+                Potion(knight_data["potion"]["name"],
+                       knight_data["potion"]["effect"])
+                if knight_data["potion"]
+                else None,
+            )
+            for knight_data in knights_config.values()
+        ]
+
+    def apply_attributes(self) -> None:
+        for knight in self.knights:
+            knight.apply_armour()
+            knight.apply_weapon()
+            knight.apply_potion()
+
+    def battle(self) -> dict[str, int]:
+        lancelot = self.knights[0]
+        mordred = self.knights[2]
+        arthur = self.knights[1]
+        red_knight = self.knights[3]
+
+        knights = [lancelot, mordred, arthur, red_knight]
+
+        for knight in range(0, len(knights), 2):
```

Comment:
def duel(knight1, knight2):
        knight2.hp -= knight1.power - knight2.protection
        knight1.hp -= knight2.power - knight1.protection

         knight1.hp = 0 if knight1.hp <= 0 else knight1.hp

Reply:
do it like this and call this function here